src/vmm/src/vstate/memory.rs

```diff
@@ -64,63 +67,198 @@ pub enum GuestRegionType {
     Hotpluggable,
 }
 
-/// An extension to GuestMemoryRegion that stores the type of region, and the KVM slot
-/// number.
+/// An extension to GuestMemoryRegion that can be split into multiple KVM slots of
+/// the same slot_size, and stores the type of region, and the starting KVM slot number.
 #[derive(Debug)]
 pub struct GuestRegionMmapExt {
     /// the wrapped GuestRegionMmap
     pub inner: GuestRegionMmap,
     /// the type of region
     pub region_type: GuestRegionType,
-    /// the KVM slot number assigned to this region
-    pub slot: u32,
+    /// the starting KVM slot number assigned to this region
+    pub slot_from: u32,
+    /// the size of the slots of this region
+    pub slot_size: usize,
```

Comment:
Is it an invariant that the slot numbers are always consecutive? If so, shall we document it explicitly in the description of the struct?